src/monodroid/jni/monodroid-glue.c

```diff
@@ -704,55 +723,104 @@ typedef void* (*mono_mkbundle_init_ptr) (void (*)(const MonoBundledAssembly **),
 mono_mkbundle_init_ptr mono_mkbundle_init;
 void (*mono_mkbundle_initialize_mono_api) (const BundleMonoAPI *info);
 
-static void
-setup_bundled_app (const char *libappso)
+static char*
+get_full_dso_path (const char *base_dir, const char *dso_path, mono_bool *needs_free)
 {
-	void *libapp;
+	if (needs_free != NULL)
```

Comment:
Alternate consideration: so that you don't need to repeat `if (needs_free != null) *needs_free=...`, you can have it point to a temporary instead:

```c
mono_bool _def_needs_free;
if (needs_free == NULL)
    needs_free = &_def_needs_free;
```

Then you can always access `*needs_free` without the check.